inventree_brother/brother_plugin.py

```diff
@@ -148,27 +147,21 @@ def print_label(self, **kwargs):
         options = kwargs.get('printing_options', {})
         n_copies = int(options.get('copies', 1))
 
-        # Look for png data in kwargs (if provided)
-        label_image = kwargs.get('png_file', None)
-
-        if not label_image:
-            # Convert PDF to PNG
-            pdf_data = kwargs['pdf_data']
-            label_image = self.render_to_png(label=None, pdf_data=pdf_data)
+        label_image = self.render_to_png(label, item)
 
         # Read settings
-        model = self.get_setting('MODEL')
-        ip_address = self.get_setting('IP_ADDRESS')
-        usb_device = self.get_setting('USB_DEVICE')
-        media_type = self.get_setting('LABEL')
+        model = machine.get_setting('MODEL', 'D')
+        ip_address = machine.get_setting('IP_ADDRESS', 'D')
+        usb_device = machine.get_setting('USB_DEVICE', 'D')
+        media_type = machine.get_setting('LABEL', 'D')
 
         # Get specifications of media type
         media_specs = None
         for label_specs in ALL_LABELS:
             if label_specs.identifier == media_type:
                 media_specs = label_specs
 
-        rotation = int(self.get_setting('ROTATION')) + 90
+        rotation = int(machine.get_setting('ROTATION', 'D')) + 90
```

Comment:
@simonkuehling this line change has started to cause some issues:

Ref: https://github.com/inventree/InvenTree/issues/8633

Obviously `D` is not a valid integer value - can you expand on what the thinking was here?

Reply:
*Following from the inventree issue.*  @SchrodingersGat  this is not a default value, the `D` means get the `ROTATION` settings from the Driver provided machine settings and not the machine type provided settings. See https://docs.inventree.org/en/stable/extend/machines/overview/#machine.BaseMachineType.get_setting

Reply:
Ok, my mistake! Still, looks like this line needs some error handling added

Reply:
Normally if everything in inventree works, this should be treated as "error handled" already, as per setting defitition a default value is specified and a int validator is applied so there is no chance that this is something different. The issue lays it inventree. We need to somehow disable the machine feature completely if no shared cache is present. Because if there is no shared cache, the machine instance is not available in the worker => the settings definition for that machine is not there, the get_setting method does not know what this setting is about => it returns "". Ideally, the `int()` in this plugin is unnecessary if everything in core works correctly.

Reply:
That makes sense. We already have the `is_global_cache_enabled` method which we use to set the number of background workers, and a few other small caching tweaks. So, that could be used here.

https://github.com/inventree/InvenTree/blob/987e0272f4a0c0e2e4751f4e6adb8ffc7102bb31/src/backend/InvenTree/InvenTree/cache.py#L29